### ISSUE TITLE
Malformed groups in reaction families: single reactant, multiple group matching often fails

### ISSUE THREAD

**Issue by connie** [ReactionMechanismGenerator/RMG-Py#386]
I noticed this error when running some sulfur radicals in an RMG-Java job.  I reran it in RMG-Py and the same error pops up. 

```
Traceback (most recent call last):
  File "../../../rmg.py", line 147, in <module>
    rmg.execute(args)
  File "/home/connie/Research/Code/RMG-Py/rmgpy/rmg/main.py", line 442, in execute
    self.initialize(args)
  File "/home/connie/Research/Code/RMG-Py/rmgpy/rmg/main.py", line 430, in initialize
    self.reactionModel.enlarge(spec)
  File "/home/connie/Research/Code/RMG-Py/rmgpy/rmg/model.py", line 620, in enlarge
    newReactions.extend(self.react(database, newSpecies))
  File "/home/connie/Research/Code/RMG-Py/rmgpy/rmg/model.py", line 568, in react
    reactionList.extend(database.kinetics.generateReactionsFromFamilies([moleculeA], products=None, failsSpeciesConstraints=self.failsSpeciesConstraints))
  File "/home/connie/Research/Code/RMG-Py/rmgpy/data/kinetics/database.py", line 419, in generateReactionsFromFamilies
    reactionList.extend(family.generateReactions(reactants, failsSpeciesConstraints=failsSpeciesConstraints))
  File "/home/connie/Research/Code/RMG-Py/rmgpy/data/kinetics/family.py", line 1215, in generateReactions
    reactionList.extend(self.__generateReactions(reactants, forward=True, failsSpeciesConstraints=failsSpeciesConstraints))
  File "/home/connie/Research/Code/RMG-Py/rmgpy/data/kinetics/family.py", line 1462, in __generateReactions
    reaction.template = self.getReactionTemplate(reaction)
  File "/home/connie/Research/Code/RMG-Py/rmgpy/data/kinetics/family.py", line 1570, in getReactionTemplate
    return self.groups.getReactionTemplate(reaction)
  File "/home/connie/Research/Code/RMG-Py/rmgpy/data/kinetics/groups.py", line 163, in getReactionTemplate
    raise UndeterminableKineticsError(reaction, message=msg)
rmgpy.data.kinetics.common.UndeterminableKineticsError: (Reaction(reactants=[Molecule(SMILES="[CH2]C1SC1C")], products=[Molecule(SMILES="CC1CC1[S]")], pairs=[[Molecule(SMILES="[CH2]C1SC1C"), Molecule(SMILES="CC1CC1[S]")]]), 'Kinetics could not be determined. Unable to find matching template for reaction <Molecule "[CH2]C1SC1C"> <=> <Molecule "CC1CC1[S]"> in reaction family <KineticsGroups "intra_substitutionCS_isomerization/groups">.Trying to match [<Entry index=1 label="XSYJ">, <Entry index=3 label="C">, <Entry index=2 label="YJ">] but matched [<Entry index=7 label="XSR4J_SS">, <Entry index=30 label="CsJ-CsHH">]')
```

The reactant is

```
multiplicity 2
1     C u0 p0 c0 {3,S} {4,S} {5,S} {11,S}
2     H u0 p0 c0 {11,S}
3     H u0 p0 c0 {1,S}
4     H u0 p0 c0 {1,S}
5     H u0 p0 c0 {1,S}
6     H u0 p0 c0 {10,S}
7     H u0 p0 c0 {10,S}
8     H u0 p0 c0 {9,S}
9  *4 C u0 p0 c0 {8,S} {10,S} {11,S} {12,S}
10 *3 C u1 p0 c0 {6,S} {7,S} {9,S}
11 *1 C u0 p0 c0 {1,S} {2,S} {9,S} {12,S}
12 *2 S u0 p2 c0 {9,S} {11,S}
```

Which needed to match these templates (these are the more generic templates for illustrative purposes)

```
XSR4J (child of XSYJ)
1 *3 R!H u1 {2,[S,D]}
2 *4 R!H u0 {1,[S,D]} {3,[S,D]}
3 *2 Ss  u0 {2,[S,D]} {4,S}
4 *1 C   u0 {3,S}

C-RRR (child of C)
1 *1 C u0 {2,S} {3,S} {4,S}
2    R u0 {1,S}
3    R u0 {1,S}
4    R u0 {1,S}

CsJ (child of YJ)
1 *3 C u1 {2,S} {3,S} {4,S}
2 *4 R u0 {1,S}
3    R u0 {1,S}
4    R u0 {1,S}
```

The reactant is able to match XSR4J and CsJ, but not C-RRR.  Even though it looks like the *1 atom in the reactant has the same connectivity.  The issue is that 2 of the atoms bonded to the *1 atom in the reactant have atomLabels, whereas the C-RRR group specifies no additional atomlabels on the bonded groups.  

In other words, the C-RRR group constrains the reactant further, and makes it so that not all subtemplates can be matched, leading to this error.   In another family, some groups constrained the number of unpaired electrons on the same reactant that matched other groups, leading to similar failures.  This seems to happen a lot in intramolecular families, or just families that require a reactant to match multiple groups.  

My fix for this is found here: https://github.com/GreenGroup/RMG-database/commit/6122415b8e994123e8eb7ab468fc0768666c3df5

The fix required creating a redfined, broader parent node for C-RRR:

```
C
1 C u0
```

I think that in the long term some sort of unit test needs to ensure that these top groups are formed correctly.  Any thoughts?

**Comment by rwest** [comment]
Is the problem that the *absence* of `*4` and `*2` labels in group `C-RRR` is being enforced so that it doesn't match the reactant?  In this particular case maybe defining `C-RRR` as
```
C-RRR
1 Cs u0
```
might do the trick, but that's not a general fix.

Certainly, more unit tests for the database are needed.
It may be worth thinking about whether *absence* of a label is enforced the same way as *presence* of a label, but someone would need to think through the consequences carefully, and check thoroughly.
